cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaVersion.java

```diff
@@ -135,43 +136,145 @@ public KafkaVersion version(String version) {
             return new TreeSet<>(map.keySet());
         }
 
-        private String image(final String crImage, final String crVersion, Map<String, String> images) {
+        private String image(final String crImage, final String crVersion, Map<String, String> images, String envVar)
+                throws NoImageException {
             final String image;
             if (crImage == null) {
                 if (crVersion == null) {
                     image = images.get(defaultVersion().version());
                     if (image == null) {
-                        throw new IllegalStateException("The images map is invalid: It lacks an image for the default version " + defaultVersion());
+                        throw new NoImageException("The images map given by " + envVar + " is invalid: " +
+                                "It lacks an image for the default version " + defaultVersion());
                     }
                 } else {
                     image = images.get(crVersion);
+                    if (image == null) {
+                        throw new NoImageException("No image for version " + crVersion +
```

Comment:
If I read the code right, this replaces the exception _version X.Y.Z is not supported_. The old message seemed to be much more clear in what the issue is. Maybe we should change it? Especially since the validation doesn't give users any option to disable some versions by not configuring the images.

Reply:
This exception now gets thrown in two contexts: 

1. During CO bootstrapping
2. During reonciliation.

The first is new and the the old exception message doesn't explain that situation very well. I'll change the code where we chain this exception into a `InvalidConfigException` or `InvalidResourceException`, and put the exception message there.